tools/generate_api_docs.py

```diff
@@ -72,6 +72,17 @@
    :nosignatures:
 
    {api_classes}
+
+Type Hints
```

Comment:
As we now define our public API via the API section (#3488) I think we should offer this guarantee to polars that there are no breaking changes in `altair.typing` within minor versions. As 5.4.0 will likely be the minimum required version, it would be great to already have it.

I fully agree with you that the docs can be made more user-friendly for these objects. However, I do not (yet) see it as a blocking issue. When I generate the docs, all type hint aliases are correctly identified by sphinx as such, nothing is identified as `TypeAliasType` as in your screenshot:

![image](https://github.com/user-attachments/assets/a60d0429-dd97-4a4b-90c6-d935c5b69869)

![image](https://github.com/user-attachments/assets/d36d6193-7820-45fc-acdb-36c23e318557)

Did you modify anything to produce this on purpose or do we have a difference? Could you maybe try again with `hatch run doc:clean-generated && hatch run doc:build-html`?

Reply:
> As we now define our public API via the API section (#3488) I think we should offer this guarantee to polars that there are no breaking changes in `altair.typing` within minor versions. As 5.4.0 will likely be the minimum required version, it would be great to already have it.

Yeah you do make a good point, great to see (#3488) mentioned!

> I fully agree with you that the docs can be made more user-friendly for these objects. However, I do not (yet) see it as a blocking issue. When I generate the docs, all type hint aliases are correctly identified by sphinx as such, nothing is identified as `TypeAliasType` as in your screenshot:

So I don't think any of them were identified with the labels (TypeAlias, TypeAliasType, ..., TypedDict) I gave in https://github.com/vega/altair/pull/3515#discussion_r1705732684. 
I used those in the comment to explain why `sphinx` had treated them differently.

> Did you modify anything to produce this on purpose or do we have a difference? Could you maybe try again with `hatch run doc:clean-generated && hatch run doc:build-html`?

Didn't modify anything to produce that intentionally.
I run on a clean build, here is another screenshot w/ another build - in case this is `sphinx` version related 

<details>
<summary>Screenshot take 2</summary>

![image](https://github.com/user-attachments/assets/2c6bd755-29d5-456f-bb25-5cd035176fef)

</details>


Maybe the issue is `hatch` isn't updating your `sphinx`, since the dependencies don't state a minimum version? 
https://github.com/vega/altair/blob/95f654fcd036069e1101883d176ec67c6f13fa25/pyproject.toml#L84-L85

Otherwise, maybe it is the `python` version we're using. I'm using `3.12`, which would be the first one to introduce [TypeAliasType](https://docs.python.org/3/library/typing.html#typing.TypeAliasType). 
Is it possible that `sphinx` would only use the symbol if built using `sys.version_info >= (3, 12)`?
That seems odd to me, as this was using `typing_extensions`.

Reply:
@binste hopefully we'll get the same `sphinx` output now I've merged this https://github.com/vega/altair/pull/3527

Reply:
Weird, even with #3527, I don't get this. I'm using Python 3.11 so maybe it's something like this...

I can build the docs with this setup when making the release. I'd suggest that if I merge this on the weekend and then make the release as-is. This makes it part of the public API and we can figure out doc issues at a later stage. Let me know if I'm missing anything, thanks! :)

Reply:
> Weird, even with #3527, I don't get this. I'm using Python 3.11 so maybe it's something like this...
> 
> I can build the docs with this setup when making the release. I'd suggest that if I merge this on the weekend and then make the release as-is. This makes it part of the public API and we can figure out doc issues at a later stage. Let me know if I'm missing anything, thanks! :)

I agree @binste, if you have consistency with `3.11` then go with that for now 👍 

Also, and this is non-blocking for me, but I'd suggest renaming the section from *Type Hints* -> *Typing*

Has been fun working with you on this

Reply:
Done 👍  And same, was fun :)